### PR TITLE
`more`: rewrite drawing logic

### PR DESCRIPTION
**Pull request by tertsdiepraam**
Should do the same as https://github.com/uutils/coreutils/pull/2380 and I restructured the code to have a `Pager` struct. I used some of the changes from #2380, but as I was already writing this, merging that first would not make sense.

Fixes/changes:
- do not accidentally skip the first line (or hide the first line)
- going to the end, back up and down again no longer skips past the last page
- state is stored in the `Pager` struct, cleaning up the main loop
- drawing the prompt and flushing are separated
- use `usize` everywhere instead of `u16`

cc @miDeb

### CODE REVIEW

File: src/uu/more/src/more.rs [hunk 1]
```diff
@@ -257,59 +235,116 @@ fn more(buff: &str, mut stdout: &mut Stdout, next_file: Option<&str>) {
...
+            self.close_on_down = false;
+        }
+        false
+    }
+
+    fn page_down(&mut self) {
+        self.upper_mark = self
+            .line_count
+            .saturating_sub(self.content_rows)
+            .min(self.upper_mark + self.content_rows);
```
Comment (miDeb) [inline src/uu/more/src/more.rs:301]: You can simply increment `self.upper_mark` by `self.content_rows` here I think.

**Comment by miDeb** [comment]
Why do we even have a `more` utility? It doesn't seem part of GNU coreutils, but of https://github.com/karelzak/util-linux :sweat_smile: It doesn't even support page-up (at least not on my version)

**Comment by tertsdiepraam** [comment]
I didn't even notice it's not part of coreutils 😄 Might as well keep it now, right? `uutils` has had `more` since 2016 it seems (https://github.com/uutils/coreutils/pull/978) and nobody brought it up then. Do you think we should maybe note that somewhere? Maybe in a `README.md` for `more`?

**Comment by miDeb** [comment]
We have to think what our goal is for this utility. Do we want to make something like `less`, or do we want to strictly look like `more`? I think we can be a bit more liberal with this utility and implement "whatever makes sense or is requested", given that we already support page-up which `more` doesn't support. Maybe rename to `less`? It looks like it is recommended to prefer `less` over `more`, so it could be better for discoverability

**Comment by tertsdiepraam** [comment]
I agree, although calling it `less` also raises expectations a bit :) We can make an issue about it and see what everyone thinks.

**Comment by hbina** [comment]
LOL! This is exactly what I am working on here https://github.com/uutils/coreutils/pull/2406
Its almost similar too.
Regarding `less` vs `more`, I recall reading somewhere that `less` streams the content while `more` reads the entire thing into memory immediately or something like that.
The manual for `more` does indicate that `less` is more advanced and should be preferred.

See [manual](https://man7.org/linux/man-pages/man1/more.1.html)

**Pull request merged**